modules/core/src/smithy4s/capability/EncoderK.scala

```diff
@@ -24,16 +24,21 @@ package smithy4s.capability
   * to pre-compile codecs for each union member, and dispatch union instances
   * to a specific codec.
   */
-trait EncoderK[F[_], Result] extends Contravariant[F] {
+trait EncoderK[F[_]] extends Contravariant[F] {
+  type Result
   def apply[A](fa: F[A], a: A): Result
   def absorb[A](f: A => Result): F[A]
   def contramap[A, B](fa: F[A])(f: B => A): F[B] =
     absorb[B](b => apply[A](fa, f(b)))
 }
 
 object EncoderK {
-  implicit def encoderKForFunction[B]: EncoderK[* => B, B] =
-    new EncoderK[* => B, B] {
+  // todo: do we even need this? Arguably you're only going to have one instance per F, and the Result is an implementation detail
```

Comment:
gotta answer this question

Reply:
users might ;)

I guess we can get rid of `Aux` and remove the type member from public signatures, if someone really needs that type they can cast while we put the types back.